lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -215,9 +245,25 @@ void TfheRustBoolEmitter::emitReferenceConversion(Value value) {
 LogicalResult TfheRustBoolEmitter::printSksMethod(
     ::mlir::Value result, ::mlir::Value sks, ::mlir::ValueRange nonSksOperands,
     std::string_view op, SmallVector<std::string> operandTypes) {
-  if (isa<TensorType>(nonSksOperands[0].getType())) {
+  // ToDo: Check again the enum definition
+  if (packedAPI && (op == "and" || op == "nand" || op == "or" || op == "nor" ||
+                    op == "xor" || op == "xnor")) {
+    std::string gateStr = std::string(op);
+    std::transform(
+        gateStr.begin(), gateStr.end(), gateStr.begin(),
+        [](unsigned char c) -> unsigned char { return std::toupper(c); });
+
     auto *opParent = nonSksOperands[0].getDefiningOp();
 
+    size_t numberOfOperands = 0;
+
+    if (isa<TensorType>(result.getType())) {
+      auto resType = mlir::dyn_cast<TensorType>(result.getType());
+      numberOfOperands = resType.getShape()[0];
```

Comment:
*just* in case this ends up being a tensor like tensor<1xN> - maybe use `resType.getNumElements()`